src/interfaces/docker_interface.py

```diff
@@ -32,94 +32,67 @@
 logging.getLogger("docker.auth").setLevel(logging.WARNING)
 
 
+def run_subprocess(command: str, config_dir: Path, ret_val: Optional[List[str]] = None) -> bool:
+    args = shlex.split(command)
+    try:
+        with subprocess.Popen(
+            args,
+            cwd=str(config_dir),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.STDOUT,
+        ) as process:
+            logger.debug(f"Running {command} with PID {process.pid}")
+            if process.stdout:
+                for line in process.stdout:
+                    decoded_line = line.decode().strip()
+                    if ret_val is not None:
+                        ret_val.append(decoded_line)
+                    else:
+                        logger.info(decoded_line)
+    except Exception as e:
+        logger.error(
+            f"An error occurred while executing `{' '.join(args)}` in {config_dir}: {e}"
+        )
+        return False
+    return True
+
+
 class DockerInterface(ContainerInterface):
     def __init__(self, network: str, config_dir: Path) -> None:
         super().__init__(network, config_dir)
         self.client = docker.DockerClient = docker.from_env()
 
     @bubble_exception_str
     def build(self) -> bool:
-        command = ["docker", "compose", "build"]
-        try:
-            with subprocess.Popen(
```

Comment:
I don't understand the commit message -- we *were* using subprocess already ...?